src/cart/crt_rpc.c

```diff
@@ -1446,6 +1446,10 @@ crt_rpc_common_hdlr(struct crt_rpc_priv *rpc_priv)
 
 	self_rank = crt_gdata.cg_grp->gg_primary_grp->gp_self;
 
+	/* If RPC failed HLC epsilon delta check return an error */
+	if (rpc_priv->crp_fail_hlc)
+		D_GOTO(out, rc = -DER_HLC_SYNC);
+
```

Comment:
On CORPC check and forwarding i agree, we should just go with early check added in corpc path to fail without forwarding to hcildren.

on a next question, if you mean here that child returns -DER_HLC_SYNC then it would be a job of the corpc aggregation function to prioritize this error code return if it detects it